### PR TITLE
Adding md40 a ckrbd compatible keymap

### PR DESCRIPTION
**Pull request by shano**
## Description

Adding a new keymap for crkbd. A minimally disruptive keymap for 40% keyboard.

## Types of Changes

- [ ] Core 
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/md40/config.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+//#define USE_MATRIX_I2C
+
+/* Select hand configuration */
+
+#define MASTER_LEFT
+// #define MASTER_RIGHT
+// #define EE_HANDS
+
+
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/config.h:31]: `MASTER_LEFT` is the default behaviour.
```suggestion
```

File: keyboards/crkbd/keymaps/md40/config.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+#define RGBLED_NUM 27
+#define RGBLIGHT_LIMIT_VAL 120
+#define RGBLIGHT_HUE_STEP 10
+#define RGBLIGHT_SAT_STEP 17
+#define RGBLIGHT_VAL_STEP 17
+
+
+#define KC_ESCC MT(MOD_LCTL, KC_ESC)
+#define KC_ENTS MT(MOD_LSFT, KC_ENT)
+#define KC_FN   MO(_FN)
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/config.h:55]:
```suggestion
```

File: keyboards/crkbd/keymaps/md40/keymap.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/keymap.c:1]: This file is missing a license header.

File: keyboards/crkbd/keymaps/md40/keymap.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/keymap.c:2]:
```suggestion
#define KC_ESCC MT(MOD_LCTL, KC_ESC)
#define KC_ENTS MT(MOD_LSFT, KC_ENT)
#define KC_FN   MO(_FN)

```

File: keyboards/crkbd/keymaps/md40/readme.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+![Base Layer](https://raw.githubusercontent.com/shano/shano.github.io/main/images/base.png)
+
+
+
+
+
+## Fn Layer
+
+![Base Layer](https://raw.githubusercontent.com/shano/shano.github.io/main/images/fn.png)
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/readme.md:23]: These images should be rehosted on Imgur.

File: keyboards/crkbd/keymaps/md40/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+SWAP_HANDS_ENABLE = no      # Enable one-hand typing
+TAP_DANCE_ENABLE = no
+
```
Comment (waffle87) [inline keyboards/crkbd/keymaps/md40/rules.mk:20]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/crkbd/keymaps/md40/keymap.c (hunk 3 above)
Comment (drashna) [reply keyboards/crkbd/keymaps/md40/keymap.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Comment (shano) [reply keyboards/crkbd/keymaps/md40/keymap.c:4]: Added, thanks!

File: keyboards/crkbd/keymaps/md40/readme.md (hunk 5 above)
Comment (shano) [reply keyboards/crkbd/keymaps/md40/readme.md]: Moved to imgur.

**Review by waffle87 — APPROVED** [review]

File: keyboards/crkbd/keymaps/md40/rules.mk [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (drashna) [inline keyboards/crkbd/keymaps/md40/rules.mk:7]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by shano** [comment]
Shall I squash all these commits now @drashna

File: keyboards/crkbd/keymaps/md40/readme.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+- Two layers only
+- Existing keyboard positions where possible
+- Exceptions where it's not possible
+- Exceptions where it makes sense
+
+Supports crkbd only.
+
+## Base Layer
+
+![Base Layer](https://i.imgur.com/hXLP6us.png)
```
Comment (drashna) [inline keyboards/crkbd/keymaps/md40/readme.md:15]:
```suggestion
![Base Layer](https://i.imgur.com/hXLP6ush.png)
```

File: keyboards/crkbd/keymaps/md40/readme.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+## Fn Layer
+
+![Base Layer](https://i.imgur.com/Sozd1p8.png)
```
Comment (drashna) [inline keyboards/crkbd/keymaps/md40/readme.md:19]:
```suggestion
![Base Layer](https://i.imgur.com/Sozd1p8h.png)
```

File: keyboards/crkbd/keymaps/md40/rules.mk [hunk 10]
```diff
@@ -0,0 +1,9 @@
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BOOTLOADER = caterina
```
Comment (drashna) [inline keyboards/crkbd/keymaps/md40/rules.mk:9]: This shouldn't be needed. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
Also, is there a specific reason for the "md40" name? 

In genreal, we like to have the keymap name to match the github user account name being used, to make things simpler for IDing.

**Comment by shano** [comment]
> Also, is there a specific reason for the "md40" name?
> 
> In genreal, we like to have the keymap name to match the github user account name being used, to make things simpler for IDing.

I can change to shano if that's preferable. md40 was simply as shortened aim of "Minimally Disruptive layout for 40% keyboards".

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**